docs/src/pages/configurations/typescript-config/index.md

```diff
@@ -0,0 +1,229 @@
+    backgroundColor: "#eee",
+    padding: "0px 5px",
+    lineHeight: "2"
+  }
+};
+export const wInfo = text => withInfo({ inline: true, source: false, styles: wInfoStyle, text: text });
+```
+
+This can be used like so:
+
+```js
+import React from "react";
+
+import { storiesOf } from "@storybook/react";
+import { PrimaryButton } from "./Button";
+import { wInfo } from "../../utils";
+import { text, select, boolean } from "@storybook/addon-knobs/react";
+
+storiesOf("Components/Button", module).addWithJSX(
+  "basic PrimaryButton",
+  wInfo(`
+
+  ### Notes
+
+  light button seen on <https://zpl.io/aM49ZBd>
+
+  ### Usage
+  ~~~js
+  <PrimaryButton
+    label={text('label', 'Enroll')}
+    disabled={boolean('disabled',false)}
+    onClick={() => alert('hello there')}
+  />
+  ~~~
+
+`)(() => (
+    <PrimaryButton
+      label={text("label", "Enroll")}
+      disabled={boolean("disabled", false)}
+      onClick={() => alert("hello there")}
+    />
+  ))
+);
+```
+
+And this is how it looks:
+
+![image](https://user-images.githubusercontent.com/35976578/38376038-ac02b432-38c5-11e8-9aed-f4fa2e258f60.png)
+
+Note: Component docgen information can not be generated for components that are only exported as default. You can work around the issue by exporting the component using a named export.
+
+## Setting up Jest tests
+
+The ts-jest [README](https://github.com/kulshekhar/ts-jest) explains pretty clearly how to get Jest to recognize TypeScript code.
+
+This is an example package.json config for jest:
+
+```json
+"jest": {
+  "transform": {
+    ".(ts|tsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js"
+  },
+  "mapCoverage": true,
+  "testPathIgnorePatterns": [
+    "/node_modules/",
+    "/lib/"
+  ],
+  "testRegex": "(/test/.*|\\.(test|spec))\\.(ts|tsx|js)$",
+  "moduleFileExtensions": [
+    "ts",
+    "tsx",
+    "js",
+    "json"
+  ]
+}
+```
+
+## Building your Typescript Storybook
+
+You will need to set up some scripts - these may help:
+
+```json
+  "scripts": {
+    "start": "react-scripts-ts start",
+    "build": "npm run lint && npm run build-lib && build-storybook",
+    "build-lib-watch": "tsc -w",
+    "build-lib": "tsc && npm run copy-css-to-lib && npm run copy-svg-to-lib && npm run copy-png-to-lib && npm run copy-woff2-to-lib",
+    "test": "react-scripts-ts test --env=jsdom",
+    "test:coverage": "npm test -- --coverage",
+    "eject": "react-scripts-ts eject",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook",
+    "copy-css-to-lib": "cpx \"./src/**/*.css\" ./build/lib",
+    "copy-woff2-to-lib": "cpx \"./src/**/*.woff2\" ./build/lib",
+    "copy-svg-to-lib": "cpx \"./src/**/*.svg\" ./build/lib",
+    "copy-png-to-lib": "cpx \"./src/**/*.png\" ./build/lib",
+    "lint": "tslint -c tslint.json 'src/**/*.{ts,tsx}'"
+  },
+```
+
+## Related Issues and Helpful Resources
+
+* [Issue: Add proper typescript webpack setup in readme #1778](https://github.com/storybooks/storybook/issues/1778)
```

Comment:
Let's leave these GitHub issues out of the docs. If you think there are useful information there, please copy and paste them to the doc instead.